viam-cartographer/src/carto_facade/carto_facade_test.cc

```diff
@@ -1440,73 +1575,187 @@ BOOST_AUTO_TEST_CASE(CartoFacade_demo_with_imu) {
-        viam_carto_lidar_reading sr = new_test_lidar_reading(
-            "lidar", ".artifact/data/viam-cartographer/mock_lidar/0.pcd",
+        add_lidar_reading_successfully(
+            vc, 1, ".artifact/data/viam-cartographer/mock_lidar/0.pcd",
             1629037851000000);
-        BOOST_TEST(viam_carto_add_lidar_reading(vc, &sr) == VIAM_CARTO_SUCCESS);
-        BOOST_TEST(viam_carto_add_lidar_reading_destroy(&sr) ==
+
+        double imu_reading[6] = {0, 0, 9.8, 0, 0, 0};
+        add_imu_reading_successfully(vc, 1, imu_reading, 1629037851000000);
+    }
+
+    // second sensor readings
+    {
+        add_lidar_reading_successfully(
+            vc, 2, ".artifact/data/viam-cartographer/mock_lidar/1.pcd",
+            1629037853000000);
+
+        double imu_reading[6] = {0.1, 0, 9.8, 0, -0.2, 0};
+        add_imu_reading_successfully(vc, 2, imu_reading, 1629037853000100);
+    }
+
+    // third sensor readings
+    {
+        add_lidar_reading_successfully(
+            vc, 3, ".artifact/data/viam-cartographer/mock_lidar/2.pcd",
+            1629037855000000);
+
+        double imu_reading[6] = {0.2, 0, 9.8, -0.6, 0, 0};
+        add_imu_reading_successfully(vc, 3, imu_reading, 1629037855000200);
+    }
+
+    // GetPosition changed from 3 successful AddLidarReading and AddIMUReading
+    // requests
+    {
+        viam_carto_get_position_response pr;
+        BOOST_TEST(viam_carto_get_position(vc, &pr) == VIAM_CARTO_SUCCESS);
+        BOOST_TEST(pr.x != 0);
+        BOOST_TEST(pr.y != 0);
+        BOOST_TEST(pr.z == 0);
+        BOOST_TEST(pr.imag != 0);
+        BOOST_TEST(pr.jmag != 0);
+        BOOST_TEST(pr.kmag != 0);
+        BOOST_TEST(pr.real != 1);
+        BOOST_TEST(to_std_string(pr.component_reference) == "lidar");
+
+        BOOST_TEST(viam_carto_get_position_response_destroy(&pr) ==
                    VIAM_CARTO_SUCCESS);
-        double readings[6] = {0, 0, 9.8, 0, 0, 0};
-        add_new_testIMUReading(vc, readings, 1629037851000000);
     }
 
-    // second sensor reading
+    // GetPointCloudMap after 3 successful sensor readings
     {
-        VLOG(1) << "viam_carto_add_lidar_reading 2";
-        viam_carto_lidar_reading sr = new_test_lidar_reading(
-            "lidar", ".artifact/data/viam-cartographer/mock_lidar/1.pcd",
-            1629037853000100);
-        BOOST_TEST(viam_carto_add_lidar_reading(vc, &sr) == VIAM_CARTO_SUCCESS);
-        BOOST_TEST(viam_carto_add_lidar_reading_destroy(&sr) ==
+        viam_carto_get_point_cloud_map_response mr;
+        BOOST_TEST(viam_carto_get_point_cloud_map(vc, &mr) ==
+                   VIAM_CARTO_SUCCESS);
+        pcl::PCLPointCloud2 blob;
+        pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud(
+            new pcl::PointCloud<pcl::PointXYZRGB>);
+        auto s = to_std_string(mr.point_cloud_pcd);
+        BOOST_TEST(viam::carto_facade::util::read_pcd(s, blob) == 0);
+        pcl::fromPCLPointCloud2(blob, *cloud);
+        BOOST_TEST(cloud != nullptr);
```

Comment:
can we have a comment explaining why no cloud is expected?

Reply:
I don't understand. It seems that a cloud is expected.

Reply:
I misread the line, apologies